lms/djangoapps/grades/management/commands/compute_grades.py

```diff
@@ -0,0 +1,122 @@
+        $ ./manage.py lms compute_grades 'edX/DemoX/Demo_Course' --settings=devstack
+    """
+    args = '<course_id course_id ...>'
+    help = 'Computes grade values for all learners in specified courses.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--courses',
+            dest='courses',
+            nargs='+',
+            help='List of (space separated) courses that need grades computed.',
+        )
+        parser.add_argument(
+            '--all_courses',
+            help='Compute grades for all courses.',
+            action='store_true',
+            default=False,
+        )
+        parser.add_argument(
+            '--routing_key',
+            dest='routing_key',
+            help='Celery routing key to use.',
+        )
+        parser.add_argument(
+            '--batch_size',
+            help='Maximum number of students to calculate grades for, per celery task.',
+            default=100,
+            type=int,
+        )
+        parser.add_argument(
+            '--start_index',
+            help='Offset from which to start processing enrollments.',
+            default=0,
+            type=int,
+        )
+
+    def handle(self, *args, **options):
+        self._set_log_level(options)
+
+        for course_key in self._get_course_keys(options):
+            self.enqueue_compute_grades_for_course_tasks(course_key, options)
+
+    def enqueue_compute_grades_for_course_tasks(self, course_key, options):
+        """
+        Enqueue celery tasks to compute and persist all grades for the
+        specified course, in batches.
+        """
+        enrollment_count = CourseEnrollment.objects.num_enrolled_in(course_key)
+        if enrollment_count == 0:
+            log.warning("No enrollments found for {}".format(course_key))
+        for offset in six.moves.range(options['start_index'], enrollment_count, options['batch_size']):
+            # If any new enrollments are added after the tasks are fired off, they are already persisting grades.
+            # so there is no need to worry about race conditions.
+            task_options = {'routing_key': options['routing_key']} if options.get('routing_key') else {}
+            result = tasks.compute_grades_for_course.apply_async(
+                kwargs={
+                    'course_key': six.text_type(course_key),
+                    'offset': offset,
+                    'batch_size': options['batch_size'],
+                },
+                options=task_options,
+            )
+            log.info("Persistent grades: Created {task_name}[{task_id}] for {course_key} [{offset}...{end}]".format(
+                task_name=tasks.compute_grades_for_course.name,
+                task_id=result.task_id,
+                course_key=course_key,
+                offset=offset,
+                end=offset + options['batch_size'],
```

Comment:
Consider DRYing this log statement by just serializing the `kwargs` passed in the line above.  This will prevent code drift in the future as well (developer adding a value to the kwarg and forgetting to update the log statement).
Also, the calculated `end` value here may not be accurate for the last batch.